### PR TITLE
load rejection sampling data + docker fix

### PR DESCRIPTION
**Pull request by jacob-morrison**
Two things in this CL:
* Adding support for our rejection sampling datasets in the data mixer
* Adding back the infiniband software bits in the docker files to fix a multinode speed issue

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,10 +35,34 @@ RUN wget https://repo.anaconda.com/miniconda/Miniconda3-py310_23.1.0-1-Linux-x86
         | sha256sum --check \
     && bash Miniconda3-py310_23.1.0-1-Linux-x86_64.sh -b -p /opt/miniconda3 \
     && rm Miniconda3-py310_23.1.0-1-Linux-x86_64.sh
-
 ENV PATH=/opt/miniconda3/bin:/opt/miniconda3/condabin:$PATH
 ENV LD_LIBRARY_PATH=/usr/local/cuda/lib:/usr/local/cuda/lib64:$LD_LIBRARY_PATH
 
+# Install a few additional utilities via pip
```
Comment (hamishivi) [inline Dockerfile:43]: Can you replicate these changes in `Dockerfile.olmo`?

File: open_instruct/utils.py [hunk 2]
```diff
@@ -278,6 +286,13 @@ def get_datasets(
                 and "messages" not in dataset.column_names
             ):
                 dataset = dataset.map(convert_metamath_qa_to_messages, num_proc=10)
+            elif (
```
Comment (hamishivi) [inline open_instruct/utils.py:289]: where is this coming from?

**Review by hamishivi — COMMENTED** [review]
Can you provide a description of the PR for memorys sake?

File: Dockerfile (hunk 1 above)
Comment (jacob-morrison) [reply Dockerfile:43]: done

File: open_instruct/utils.py (hunk 2 above)
Comment (jacob-morrison) [reply open_instruct/utils.py:289]: this was the original point of this branch, adding support for loading our rejection sampling datasets -- they're both small changes so I didn't bother splitting them. I already confirmed this works + it fits with our general schema for loading non-messages-oriented datasets

File: Dockerfile (hunk 1 above)
Comment (hamishivi) [reply Dockerfile:43]: I dont see the `gpustat`, `yes | unminimize` etc changes? The two dockerfiles should be more or less identical minus the diff requirement files.

File: open_instruct/utils.py (hunk 2 above)
Comment (hamishivi) [reply open_instruct/utils.py:289]: would be good to document this in PR description :)

File: Dockerfile (hunk 1 above)
Comment (jacob-morrison) [reply Dockerfile:43]: oh good catch, incomplete copy/paste. they should line up now

File: open_instruct/utils.py (hunk 2 above)
Comment (jacob-morrison) [reply open_instruct/utils.py:289]: was just about to :) done!

**Review by hamishivi — APPROVED** [review]

**Pull request merged**